core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/CloudStorageDecorator.java

```diff
@@ -46,19 +65,75 @@ public CloudStorageRequest decorate(CloudStorageRequest request, DetailedEnviron
                     identities.add(identity);
                 }
             }
+            List<SdxClusterResponse> datalakes = sdxClientService.getByEnvironmentCrn(environment.getCrn());
+            updateCloudStorageLocations(blueprintName, clusterName, request, datalakes);
+            updateDynamoDBTable(request, environment);
+        }
+        return request;
+    }
 
-            if (dynamoDBTableNameSpecified(environment)) {
-                String dynamoDbTableName = environment.getAws().getS3guard().getDynamoDbTableName();
-                S3Guard s3Guard = new S3Guard();
-                s3Guard.setDynamoTableName(dynamoDbTableName);
-                AwsStorageParameters aws = new AwsStorageParameters();
-                aws.setS3Guard(s3Guard);
-                request.setAws(aws);
+    public void updateDynamoDBTable(CloudStorageRequest request, DetailedEnvironmentResponse environment) {
+        if (dynamoDBTableNameSpecified(environment)) {
+            String dynamoDbTableName = environment.getAws().getS3guard().getDynamoDbTableName();
+            S3Guard s3Guard = new S3Guard();
+            s3Guard.setDynamoTableName(dynamoDbTableName);
+            AwsStorageParameters aws = new AwsStorageParameters();
+            aws.setS3Guard(s3Guard);
+            request.setAws(aws);
+        }
+    }
+
+    public CloudStorageRequest updateCloudStorageLocations(String blueprintName, String clusterName,
+        CloudStorageRequest request, List<SdxClusterResponse> datalakes) {
+        if (hasDatalake(datalakes) && storageLocationsNotDefined(request)) {
+            if (request == null) {
+                request = new CloudStorageRequest();
+            }
+            SdxClusterResponse sdxClusterResponse = datalakes.get(0);
+            String baseLocation = sdxClusterResponse.getCloudStorageBaseLocation();
+            FileSystemType fileSystemType = sdxClusterResponse.getCloudStorageFileSystemType();
+            Set<ConfigQueryEntry> recommendations = new HashSet<>();
+            if (!Strings.isNullOrEmpty(baseLocation) && fileSystemType != null) {
+                recommendations = getFileSystemRecommendations(blueprintName, clusterName, baseLocation, fileSystemType.name());
+            }
+            if (request.getLocations() == null) {
+                request.setLocations(List.of());
+            }
+            for (ConfigQueryEntry recommendation : recommendations) {
+                StorageLocationBase storageLocationBase = new StorageLocationBase();
+                storageLocationBase.setType(recommendation.getType());
+                storageLocationBase.setValue(recommendation.getDefaultPath());
+                request.getLocations().add(storageLocationBase);
```

Comment:
if the line#100 is called this will throw an exception because this is an immutable list:
`static <E> List<E> of() {
        return ImmutableCollections.emptyList();
    }
`

please write a unit test for it